frontend/ui/wikipedia.lua

```diff
@@ -502,20 +516,23 @@ function Wikipedia:addImages(page, lang, more_images, image_size_factor, hi_imag
         end
         width = math.ceil(width * image_size_factor)
         height = math.ceil(height * image_size_factor)
+        -- Ask for the next largest thumbnail, and then rely on scaling down.
+        local request_width = round_to_thumbnail_step(width)
         -- All wikipedia image urls like .../wikipedia/commons/A/BC/<filename>
         -- or .../wikipedia/commons/thumb/A/BC/<filename>/<width>px-<filename>
         -- can be transformed to another url with a requested new_width with the form:
         --   /wikipedia/commons/thumb/A/BC/<filename>/<new_width>px-<filename>
         -- (Additionally, the image format can be changed by appending .png,
         -- .jpg or .gif to it)
-        -- The resize is so done on Wikipedia servers from the source image for
-        -- the best quality.
-        local source = wimage.source:gsub("(.*/)%d+(px-[^/]*)", "%1"..width.."%2")
+        -- Round to Wikipedia's thumbnail steps to avoid 429 errors
+        local source = wimage.source:gsub("(.*/)%d+(px-[^/]*)", "%1"..request_width.."%2")
```

Comment:
Please move your 2 first added lines with the other ones, it doesn't need to be that far away :) and we can make the comment more consistent:
```lua
        -- Since 20251225, Wikipedia requires that we request a thumbnail with a width from a fixed set
        -- of images sizes (or we would get a HTTP 429 error).
        local requested_width = round_to_thumbnail_step(width)
        local source = wimage.source:gsub("(.*/)%d+(px-[^/]*)", "%1"..requested_width.."%2")
```
Also, `request_width` sounds like a boolean, so may be use `requested_width` ?